pagopa/io-app
# [IOCOM-1675, IOCOM-1677] Handling of required App Update for FIMS

⚠️ This PR depends on [406](https://github.com/pagopa/io-dev-api-server/pull/406) ⚠️

## Short description
This PR adds the app version check and update required on FIMS for both authentication and history screens.
|Auth Screen|History Screen|
|-|-|
|![Simulator Screenshot - iPhone 15 - 2024-08-06 at 11 46 15](https://github.com/user-attachments/assets/c8b00990-2cc1-4fc7-adcb-8577595c2cba)|![Simulator Screenshot - iPhone 15 - 2024-08-06 at 11 45 41](https://github.com/user-attachments/assets/c870ae3b-6d6e-4d1c-ae10-27228c284f89)|

## List of changes proposed in this pull request
- `io-services-metadata` specs updated to `1.0.39`
- `fimsRequiresAppUpdateSelector` that returns `true` if an app update for FIMS is required
- `FimsFlowHandlerScreen` and `HistoryScreen` use the selector above to display an `OperationResultScreenContent` that prevents the user from using both authentication and history features. Such selector output is also used to prevent the starting of related networking requests.
- Be aware that a proper UI/UX will come later

## How to test
Using the io-dev-api-server, change the `fims.min_app_version` property to simulate both cases and check that the app behaves properly.
